addons/mrp/models/product.py

```diff
@@ -70,10 +73,18 @@ class ProductProduct(models.Model):
         compute='_compute_used_in_bom_count', compute_sudo=False)
     mrp_product_qty = fields.Float('Manufactured',
         compute='_compute_mrp_product_qty', compute_sudo=False)
+    is_kits = fields.Boolean(compute="_compute_bom_count", compute_sudo=False)
+
+    @api.depends('bom_ids')
+    def _compute_is_kits(self):
+        for p in self:
+            p.is_kits = any(b.type == "phantom" for b in p.variant_bom_ids)
 
     def _compute_bom_count(self):
         for product in self:
```

Comment:
Improve perf (if product template have 1000 boms)
```python
    def _compute_bom_count(self):
        for product in self:
            domain = ['|', ('product_id', '=', product.id), '&', ('product_id', '=', False), ('product_tmpl_id', '=', product.product_tmpl_id.id)]
            product.bom_count = self.env['mrp.bom'].search_count(domain)
            product.is_kits = bool(self.env['mrp.bom'].search_count(domain + [('type','=','phantom')]))
```

Reply:
Batched it completely